scipy/stats/morestats.py

```diff
@@ -2078,6 +2078,54 @@ def anderson_ksamp(samples, midrank=True):
 AnsariResult = namedtuple('AnsariResult', ('statistic', 'pvalue'))
 
 
+
+class _ABW:
+    '''Distribution of Ansari-Bradley W-statistic under the null hypothesis'''
+    # TODO: calculate exact distribution considering ties
+    # We could avoid summing over more than half the frequencies,
+    # but inititally it doesn't seem worth the extra complexity
```

Comment:
Easy to add, though: rename the `sf` and `cdf` functions with leading underscores to make them private. The public `sf`, for example, can choose to use `_sf` or `1 - _cdf`.

Reply:
> Easy to add, though: rename the `sf` and `cdf` functions with leading underscores to make them private. Then the public `sf`, for example, can choose to use `_sf` or `1 - _cdf`.

Sorry, I don't get this (as I am not familiar with the ties correction method in exact distribution)! How would this help with ties?

Reply:
It won't do anything for ties, just possibly accuracy. Currently in the main PR, you calculate left and right P values from different tails depending on whether the statistic is to the left or right of the median. I figured this was for accuracy. I don't do that here, but it would be easy to add.

Reply:
Oh, I get it now. The placement of the comment just under the TODO comment confused me into believing this had to do something with ties. Never mind.

I agree that the extra complexity will make it harder to understand the code. So, it's probably OK to not have it.